src/audio/eq_iir/CMakeLists.txt

```diff
@@ -1,3 +1,3 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-add_local_sources(sof eq_iir.c iir.c)
+add_local_sources(sof eq_iir.c iir.c iir_hifi3.c)
```

Comment:
Maybe we can add this "#if IIR_GENERIC" here instead of compiling two files?

Reply:
I'm rather novice with cmake. Is such used elsewhere in project as example?

Reply:
Yes, for example this one is a huge CMakeLists:
src\audio\CMakeLists.txt

example:

```
	if(CONFIG_COMP_MIXER)
		add_local_sources(sof
			mixer.c
		)
	endif()
```

Reply:
Actually src and volume should be done in the same way, there is no point in compiling hifi3 file if there is no hifi3.
Problem is that you need to know if platform supports f.e. HIFI3 to show this config at all and you won't look up into xtensa core headers cos things like XCHAL_HAVE_HIFI3 are not visible in kconfig phase. So you would have to duplicate this in kconfig entries like `HAVE_HIFI3` what I'd like not to do...

Maybe it's better to by default search for best possible implementation (like it's now) and have config's like FORCE_VOLUME_GENERIC if someone doesn't want f.e. hifi3 accelerated component for some reason.

Reply:
Thats right, in another comment I suggested kconfig options GENERIC and AUTO (so still decided in header file which version to use like now) so it will be possible in cmake to compile one (generic) file or both of them.